sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1458,6 +1458,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val JSON_EXPRESSION_OPTIMIZATION =
+    buildConf("spark.sql.optimizer.jsonExpressionOptimization")
```

Comment:
`Optimization` sounds a bit redundant with `optimizer`. I wonder if we have a better name ..  I am okay with as-is if we don't have a better idea.